pyproject.toml

```diff
@@ -23,61 +25,37 @@ classifiers = [# classifiers for license, versions set automatically during Poet
 packages = [
     { include = "kolena" }
 ]
+requires-python = ">=3.8,<3.13"
 
-[tool.poetry.dependencies]
-python = ">=3.8,<3.13"
-numpy = [
-    { version = ">=1.19,<2", python = ">=3.8,<3.11" },
-    { version = ">=1.23,<2", python = ">=3.11,<3.12" },
-    { version = ">=1.26,<2", python = ">=3.12" },
-]
-pandas = [
-    { version = ">=1.1,<3", python = ">=3.8,<3.11" },
-    { version = ">=1.5,<3", python = ">=3.11,<3.12" },
-    { version = ">=2.1.1,<3", python = ">=3.12" },
+dependencies = [
+    "numpy >=1.19,<2; python_version < '3.12'",
+    "numpy >=1.26,<2; python_version >= '3.12'",
+    "pandas >=1.1,<3; python_version < '3.12'",
+    "pandas >=2.1.1,<3; python_version >= '3.12'",
+    "pandera>=0.9,<1",
+    "pydantic>=1.10,<3",
+    "dacite>=1.6,<2",
+    "requests>=2.20,<3",
+    "requests-toolbelt>=1.0.0,<2",
+    "tqdm>=4,<5",
+    "Pillow>=10.0.1,<11",
+    "retrying>=1.3.3,<2",
+    "Shapely>=1.8.5,<3",
+    "termcolor>=1.1,<3",
+    "pyarrow>=8",
+    "click>=8",
+    "lzstring>=1.0.4,<2",
 ]
-pandera = ">=0.9,<1"
-pydantic = ">=1.10,<3"
-dacite = ">=1.6,<2"
-requests = ">=2.20,<3"
-requests-toolbelt = "*"
-tqdm = ">=4,<5"
-Pillow = ">=10.0.1,<11"
-retrying = ">=1.3.3,<2"
-Shapely = ">=1.8.5,<3"
-termcolor = ">=1.1,<3"
-pyarrow = ">=8"
-click = ">=8"
-scikit-learn = { version = ">=1.2,<2", optional = true }
-scipy = [  # transient dependency of scikit-learn, pin required to unbreak installs on specific Python versions
-    { version = ">=1,<1.11", python = ">=3.8,<3.9", optional = true },
-    { version = ">=1,<1.14", python = ">=3.9,<3.10", optional = true },
-    { version = ">=1.12,<2", python = ">=3.12", optional = true }
-]
-lzstring = "^1.0.4"
-
-[tool.poetry.extras]
-metrics = ["scikit-learn", "scipy"]
 
-[tool.poetry.group.dev.dependencies]
-pre-commit = "^2.17"
-black = { version = "^22.1.0", allow-prereleases = true }
-pytest = "^7"
-pytest-cov = "^4.0.0"
-pytest-depends = "^1.0.1"
-mkdocs = "^1.4.3"
-cairosvg = "^2.7.0"
-mkdocs-material = ">=9.2,<10"  # insiders fork installed out-of-band in docs/setup_insiders.sh
-mkdocstrings = { version = ">0.20,<1", extras = ["python"] }
-mkdocs-git-committers-plugin-2 = "^1"
-mkdocs-git-revision-date-localized-plugin = "^1"
-
-[tool.poetry.scripts]
-kolena = 'kolena._utils.cli:run'
```

Comment:
i think we need to find the counterpart of uv, b/c it provides a kolena cli in the python venv after installing kolena package

Reply:
@yifanwu-kolena what is this used for / who uses it (is it something publicly shared / documented)? I couldn't find a great counterpart, and I'm not sure it's worth blocking this based on the cli.

Reply:
Ah, I see: https://docs.kolena.com/workflow/advanced-usage/packaging-for-automated-evaluation/

Reply:
Actually, this seems to work out of the box with uv!
```
❯ uv run kolena --help
Ignoring existing lockfile due to change in supported environments
Usage: kolena [OPTIONS] COMMAND [ARGS]...

Options:
  -h, --help  Show this message and exit.

Commands:
  evaluator
  repository
```

Reply:
this is cool, but how can uv know the cli entrypoint? 🤔 

based on [the doc](https://github.com/astral-sh/uv/blob/039b68367e6481690ca001c44e61500bd1e42570/docs/concepts/projects.md?plain=1#L210-L255), we can define `[project.scripts]`

Reply:
Hmm, I wonder if there was something leftover from my local state with `[tool.poetry.scripts]`

Nice catch, re-added as `[project.scripts]`

Reply:
maybe there are some heuristic magic behind the scene, but i believe explicit is better than implicit